api/features/serializers.py

```diff
@@ -244,6 +256,87 @@ class Meta:
     identity = _IdentitySerializer()
 
 
+class FeatureStateValueEdgeIdentityField(serializers.Field):
+    def to_representation(self, obj):
+        identity_id = self.parent._get_identity_uuid()
+        return obj.get_value(identity_id=identity_id)
+
+    def get_attribute(self, instance):
+        return instance
+
+    def to_internal_value(self, data):
+        feature_state_value_dict = FeatureState().generate_feature_state_value_data(
+            data
+        )
+
+        data = {**feature_state_value_dict}
+        fs_value_serializer = FeatureStateValueSerializer(data=data)
+        fs_value_serializer.is_valid(raise_exception=True)
+        return FeatureStateValue(**data).value
+
+
+class EdgeFeatureField(serializers.Field):
+    def to_representation(self, obj):
+        return obj.id
+
+    def to_internal_value(self, data):
+        feature = Feature.objects.get(id=data)
+        return feature
+
+
+class FeatureStateSerializerWithEdgeIdentity(serializers.Serializer):
+    feature_state_value = FeatureStateValueEdgeIdentityField()
+    feature = EdgeFeatureField()
+    multivariate_feature_state_values = EdgeMultivariateFeatureStateValueSerializer(
+        many=True, required=False
+    )
+    enabled = serializers.BooleanField(required=False, default=False)
+    identity_uuid = serializers.SerializerMethodField()
+
+    featurestate_uuid = serializers.CharField(required=False, read_only=True)
+
+    def _get_identity_uuid(self):
+        return self.context["view"].kwargs["edge_identity_identity_uuid"]
+
+    def get_identity_uuid(self, obj):
+        return self._get_identity_uuid()
+
+    def save(self, **kwargs):
+        identity = self.context["view"].identity
+        feature_state_value = self.validated_data.pop("feature_state_value")
+
+        if self.instance:
+            if self.validated_data.get("multivariate_feature_state_values"):
+                engine_multi_fs_value_models = engine_multi_fs_value_schema.load(
+                    self.validated_data["multivariate_feature_state_values"], many=True
+                )
+                self.instance.multivariate_feature_state_values = (
+                    engine_multi_fs_value_models
+                )
+            self.instance.set_value(feature_state_value)
```

Comment:
Since this is in the `if` and the `else` clause, couldn't we just move it below? Before the `try` / `except` ?